### PR TITLE
[DS-3762] fixed test for subCommunities

### PR DESCRIPTION
**Pull request by Raf-atmire**
This PR fixes the test on the /communities/search/subCommunities?parent=UUID endpoint as described by @abollini. cfr: https://github.com/DSpace/DSpace/pull/1893

### CODE REVIEW

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java [hunk 1]
```diff
@@ -253,17 +252,23 @@ public void findAllSubCommunities() throws Exception{
                 .withName("Sub Community")
                 .build();
 
-        Community child12 = CommunityBuilder.createSubCommunity(context, child1)
+        Community child12 = CommunityBuilder.createSubCommunity(context, parentCommunity)
```
Comment (tomdesair) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java]: @Raf-atmire The naming suggests that this Community is a sub-community of `child1`, but it's not. Can you rename this variable to `child2`?

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java [hunk 2]
```diff
@@ -273,11 +278,40 @@ public void findAllSubCommunities() throws Exception{
                 .andExpect(jsonPath("$._embedded.communities", Matchers.not(Matchers.containsInAnyOrder(
```
Comment (tomdesair) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java]: 1. Please add a comment here that you are checking that the following communities are not present.
2. You should not use `containsInAnyOrder` to check something is not present. You should use
```
not(anyOf(
   hasItem(...),
   hasItem(...),
   hasItem(...) 
))
```

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java [hunk 3]
```diff
@@ -253,17 +252,23 @@ public void findAllSubCommunities() throws Exception{
...
+                .withName("Sub Community2")
+                .build();
+
+        Community child121 = CommunityBuilder.createSubCommunity(context, child12)
                 .withName("Sub Sub Community")
                 .build();
 
+
         Community child2 = CommunityBuilder.createSubCommunity(context, parentCommunity2)
                 .withName("Sub2 Community")
```
Comment (tomdesair) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java]: This would be sub-community 3.

**Review by tomdesair — CHANGES_REQUESTED** [review]
I just have some minor remarks to make the tests more clear.

**Review by tomdesair — APPROVED** [review]

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java [hunk 4]
```diff
@@ -249,35 +248,74 @@ public void findAllSubCommunities() throws Exception{
...
+
+        Community parentCommunity2Child1 = CommunityBuilder.createSubCommunity(context, parentCommunity2)
                 .withName("Sub2 Community")
                 .build();
 
-        Collection col1 = CollectionBuilder.createCollection(context, child1).withName("Collection 1").build();
+        Collection col1 = CollectionBuilder.createCollection(context, parentCommunityChild1).withName("Collection 1").build();
 
-        getClient().perform(get("/api/core/communities/search/subCommunities/" + parentCommunity.getID()))
+        getClient().perform(get("/api/core/communities/search/subCommunities/")
```
Comment (abollini) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java]: this is unexpected... the URL should be /api/core/communities/search/subCommunities without the final /
should we reject such request?

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java [hunk 5]
```diff
@@ -249,35 +248,74 @@ public void findAllSubCommunities() throws Exception{
...
+                        CommunityMatcher.matchCommunityEntry(parentCommunity2Child1.getName(), parentCommunity2Child1.getID(), parentCommunity2Child1.getHandle()),
+                        CommunityMatcher.matchCommunityEntry(parentCommunityChild2Child1.getName(), parentCommunityChild2Child1.getID(), parentCommunityChild2Child1.getHandle())
                 ))))
                 .andExpect(jsonPath("$._links.self.href", Matchers.containsString("/api/core/communities/search/subCommunities")))
                 .andExpect(jsonPath("$.page.size", is(20)))
-                .andExpect(jsonPath("$.page.totalElements", is(3)))
+                .andExpect(jsonPath("$.page.totalElements", is(2)))
+        ;
+
+        getClient().perform(get("/api/core/communities/search/subCommunities/")
```
Comment (abollini) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java]: without the final /

File: dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java [hunk 6]
```diff
@@ -249,35 +248,74 @@ public void findAllSubCommunities() throws Exception{
...
+                        CommunityMatcher.matchCommunityEntry(parentCommunity2Child1.getName(), parentCommunity2Child1.getID(), parentCommunity2Child1.getHandle()),
+                        CommunityMatcher.matchCommunityEntry(parentCommunityChild2Child1.getName(), parentCommunityChild2Child1.getID(), parentCommunityChild2Child1.getHandle()),
+                        CommunityMatcher.matchCommunityEntry(parentCommunityChild1.getName(), parentCommunityChild1.getID(), parentCommunityChild1.getHandle())
+                ))))
+                .andExpect(jsonPath("$._links.self.href", Matchers.containsString("/api/core/communities/search/subCommunities")))
+                .andExpect(jsonPath("$.page.size", is(20)))
+                .andExpect(jsonPath("$.page.totalElements", is(1)))
+        ;
+
+        getClient().perform(get("/api/core/communities/search/subCommunities/")
```
Comment (abollini) [inline dspace-spring-rest/src/test/java/org/dspace/app/rest/CommunityRestRepositoryIT.java]: without the final /

**Review by abollini — CHANGES_REQUESTED** [review]

**Review by abollini — APPROVED** [review]

**Pull request merged**